apps/docker-base/Makefile.docker-base

```diff
@@ -0,0 +1,33 @@
+# -*- makefile -*-
+help:
+	@echo "This should be run as part of the Dockerfile"
+	false
+
+user-setup:
+	groupadd --gid 1000 app
+	useradd -d /app --uid 1000 --gid app app
+	chown -R app:app /app
+
+apt-setup:
+	rm -f /etc/apt/apt.conf.d/docker-clean
+	echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+
+apt-install:
+	DEBIAN_FRONTEND=noninteractive apt update
+	# gcc and python3-dev needed on arm for guidance
```

Comment:
On arm, yes, on x86 no.  I had initially removed these when I rewrote to the makefiles because I couldn't figure out why they were needed.  Then I built the arm images and it failed.